src/v/cluster/partition_probe.cc

```diff
@@ -39,6 +40,70 @@ void replicated_partition_probe::setup_metrics(const model::ntp& ntp) {
       partition_label(ntp.tp.partition()),
     };
 
+    _metrics.add_group(
+      prometheus_sanitize::metrics_name("kafka"),
+      {
+        // Partition Level Metrics
+        sm::make_gauge(
+          "max_offset",
+          [this] {
+              auto log_offset = _partition.committed_offset();
+              auto translator = _partition.get_offset_translator_state();
+
+              return translator->from_log_offset(log_offset);
+          },
+          sm::description("Partition max offset"),
+          labels,
+          {sm::shard_label}),
+        sm::make_gauge(
+          "under_replicated_replicas",
+          [this] {
+              auto metrics = _partition._raft->get_follower_metrics();
+              return std::count_if(
+                metrics.cbegin(),
+                metrics.cend(),
+                [](const raft::follower_metrics& fm) {
+                    return fm.under_replicated;
+                });
+          },
+          sm::description("Number of under replicated replicas"),
+          labels,
+          {sm::shard_label}),
+        // Topic Level Metrics
+        sm::make_total_bytes(
+          "request_bytes_total",
+          [this] { return _bytes_produced; },
+          sm::description("Total number of bytes produced per topic"),
+          {request_label("produce"),
+           ns_label(ntp.ns()),
+           topic_label(ntp.tp.topic()),
+           partition_label(ntp.tp.partition())},
+          sm::impl::shard(),
+          {sm::shard_label, partition_label}),
+        sm::make_total_bytes(
+          "request_bytes_total",
+          [this] { return _bytes_fetched; },
+          sm::description("Total number of bytes consumed per topic"),
+          {request_label("consume"),
+           ns_label(ntp.ns()),
+           topic_label(ntp.tp.topic()),
+           partition_label(ntp.tp.partition())},
+          sm::impl::shard(),
+          {sm::shard_label, partition_label}),
+        sm::make_gauge(
+          "replicas",
+          [this] {
+              auto metrics = _partition._raft->get_follower_metrics();
```

Comment:
I wonder if it's worth adding a function to get the count without all of the metrics. Perhaps: `_partition._raft->get_follower_count()`